src/sentry/search/events/fields.py

```diff
@@ -2309,28 +2309,40 @@ def __init__(self, dataset: Dataset, params: ParamsType):
         }
 
     def resolve_select(self, selected_columns: Optional[List[str]]) -> List[SelectType]:
+        """Given a public list of discover fields, construct the corresponding
+        list of Snql Columns or Functions. Duplicate columns are ignored
+        """
         if selected_columns is None:
             return []
 
         columns = []
 
-        for field in selected_columns:
-            if field.strip() == "":
+        for column in selected_columns:
+            if column.strip() == "":
                 continue
-            resolved_field = self.resolve_field(field)
-            if resolved_field not in self.columns:
-                columns.append(resolved_field)
+            resolved_column = self.resolve_column(column)
+            if resolved_column not in self.columns:
+                columns.append(resolved_column)
 
         return columns
 
-    def resolve_field(self, field: str) -> SelectType:
+    def resolve_column(self, field: str) -> SelectType:
+        """Given a public discover field, construct the corresponding Snql, this
+        function will determine the type of the field alias, whether its a
+        column, field alias or function and call the corresponding resolver
+        """
         match = is_function(field)
         if match:
             return self.resolve_function(field, match)
-
-        if self.is_field_alias(field):
+        elif self.is_field_alias(field):
             return self.resolve_field_alias(field)
+        else:
+            return self.resolve_field(field)
 
+    def resolve_field(self, field: str) -> Column:
+        """Given a public discover field, resolve the alias based on the Query's
+        dataset and return the Snql Column
+        """
```

Comment:
Hm... doesn't a public discover field imply we're working with the discover dataset?

Reply:
hmmm does `public dataset field` read better?

Reply:
went with just removing references to `discover` so just `public field` 🤷